Assets/Scripts/Audio/AudioManager.cs

```diff
@@ -0,0 +1,120 @@
+using System.Collections;
+using System.Collections.Generic;
+using System.IO;
+using UnityEngine;
+
+public class AudioManager : MonoBehaviour
+{
+    // Singleton pattern
+    public static AudioManager Instance { get; private set; }
+    private void Awake()
+    {
+        if (Instance != null && Instance != this)
+        {
+            Destroy(this);
+        }
+        else
+        {
+            Instance = this;
+            DontDestroyOnLoad(this);
+        }
+    }
```

Comment:
Instead of the Singleton pattern, we can just bind a single AudioManager instance to the Extenject Dependency Injection Project or Scene Container (https://github.com/Mathijs-Bakker/Extenject), from which it can be accessed by any object in the game. We should aim to use the DI patterns described in Extenject for the game. See ProjectContextInstaller.cs for a very limited example of this in the game.